src/python_testing/TC_OPCREDS_3_8.py

```diff
@@ -735,7 +733,131 @@ async def test_TC_OPCREDS_3_8(self):
             asserts.assert_true(verify_signature(public_key=noc_public_keys_from_certs["TH2"], message=vendor_id_verification_tbs,
                                 signature=sign_vid_verification_response.signature), "VID Verification Signature must validate using DUT's NOC public key")
 
-        with test_step(22, description="Remove TH2 and TH3's fabric"):
+        # Cover having done UpdateNOC in a Fail-safe, overriding the VVS and VVSC, and then expiring failsafe and seeing the prior values are returned.
+        with test_step(22, description="Start updating fabric under TH2's root with fabric ID 0x2222, node ID 0x22222222, by invoking ArmFailSafe(600s), CSRRequest and UpdateNOC. Do not disarm failsafe, do not execute commissioning complete."):
```

Comment:
@cecille Leaving this comment open until Tuesday to give you a chance to review.